### PR TITLE
Flush output buffers before rendering story

### PR DESCRIPTION
**Pull request by swissspidy**
Quick temporary workaround for #967.

### CODE REVIEW

File: includes/templates/single-web-story.php [hunk 1]
```diff
@@ -26,7 +28,12 @@
...
 
-?>
-<!DOCTYPE html>
-<?php
-the_content();
+$post = get_post();
+
+if ( $post instanceof WP_Post ) {
+	// TODO: Fix issue with AMP plugin.
+	ob_end_clean();
```
Comment (swissspidy) [inline includes/templates/single-web-story.php]: Removing this breaks things.

Need to ensure the AMP plugins output buffer runs after ours.

**Comment by spacedmonkey** [comment]
@swissspidy Can you explain the detail a little more so I can easily understand what is going and why it needed? I dont feel like I have any context to what I am seeing here.

**Comment by swissspidy** [comment]
@spacedmonkey No need to review just now (it's WIP). I'd like to await feedback from the AMP plugin folks on this first. But see #967 for background. Basically there's nothing (or just broken markup) output on the frontend when the AMP plugin is active. Trying to fix it.

**Comment by swissspidy** [comment]
Actually, treating this as an urgent temp fix for alpha.

@spacedmonkey @miina can you have a quick look? This PR just simplifies rendering (no need to go through `the_content()`) and now calls `ob_end_clean()` before that.

**Review by westonruter — CHANGES_REQUESTED** [review]
I recommend this instead: https://github.com/google/web-stories-wp/issues/967#issuecomment-607989846

**Comment by swissspidy** [comment]
Updated, thanks @westonruter

File: includes/Story_Post_Type.php [hunk 2]
```diff
@@ -192,6 +173,19 @@ static function ( $content ) {
 		add_action( 'web_stories_story_head', 'wp_shortlink_wp_head', 10, 0 );
 		add_action( 'web_stories_story_head', 'wp_site_icon', 99 );
 		add_action( 'web_stories_story_head', 'wp_oembed_add_discovery_links' );
+
+		// @todo Improve AMP plugin compatibility, see https://github.com/google/web-stories-wp/issues/967
+		add_filter(
+			'amp_skip_post',
+			static function( $skipped, $post ) {
+				if ( 'web-story' === get_post_type( $post ) ) {
```
Comment (westonruter) [inline includes/Story_Post_Type.php]:
```suggestion
				if ( self::POST_TYPE_SLUG === get_post_type( $post ) ) {
```

File: includes/templates/single-web-story.php [hunk 3]
```diff
@@ -26,7 +28,10 @@
...
-?>
-<!DOCTYPE html>
-<?php
-the_content();
+$post = get_post();
+
+if ( $post instanceof WP_Post ) {
+	echo '<!DOCTYPE html>';
+	$renderer = new Story_Renderer( $post );
+	echo $renderer->render();
```
Comment (westonruter) [inline includes/templates/single-web-story.php:36]: Should tabs be used here?

**Review by westonruter — APPROVED** [review]

File: includes/templates/single-web-story.php (hunk 3 above)
Comment (swissspidy) [reply includes/templates/single-web-story.php:36]: Why not? 🤔 

We use spaces for JS, tabs for PHP.

Comment (westonruter) [reply includes/templates/single-web-story.php:36]: Sorry, the GitHub diff view made it look like spaces were used.

**Pull request merged**